src/compiler/scala/tools/nsc/transform/patmat/Logic.scala

```diff
@@ -309,10 +309,8 @@ trait Logic extends Debugging  {
 
     }
 
-    // TODO: remove since deprecated
-    val budgetProp = scala.sys.Prop[String]("scalac.patmat.analysisBudget")
-    if (budgetProp.isSet) {
-      reportWarning(s"Please remove -D${budgetProp.key}, it is ignored.")
+    if (System.getProperty("scalac.patmat.analysisBudget") != null) {
```

Comment:
That's where all the `Properties.{propOrXXX, propIsSet, ...}` methods are useful. I hadn't even realized it was all duplicated in `sys.props` until I saw it being used in the REPL.

Reply:
I would just re-read the source whenever I needed to use it.